### ISSUE TITLE
on mouseup - complete on Return insert some strange text

### ISSUE THREAD

**Issue by matthiasrebbe** [mwieder/glx2ScriptEditor#3]
When pressing return after **on mouseonp**  not only **end mouseup** is added, but also this text
   _-- replace this with your own code
   throw param(0) && "has not yet been implemented"_

can this be changed somewhere or is coded in the script?

**Comment by mwieder** [comment]
There's a preference for Test-driven Development on the preferences panel. I thought I had the default set to false, but apparently that's not so. Check the checkbox, then uncheck it again and it should be turned off. I'll put changing the default on my todo list.

**Comment by mwieder** [comment]
Closed by accident.

**Comment by mwieder** [comment]
Again. Sheesh.

**Comment by matthiasrebbe** [comment]
That is not working. Did as you suggested. Checked, unchecked that "test-driven.." option. Still getting that text after pressing Return when "complete on Return" is enabled.

**Comment by mwieder** [comment]
Well, that's disturbing - it does work here, but I don't suppose that helps you much. Is there an entry near the bottom of your "GLX2 Code Prefs.txt" file for "uEncourageTDD", and is the value false?

**Comment by mwieder** [comment]
I just looked through the code, and the default is indeed "false", so you shouldn't have that happen unless the value explicitly gets set.

**Comment by matthiasrebbe** [comment]
This is the entry
`<uEncourageTDD>false</uEncourageTDD>`, so it seems to be correct.
But it still shows the behavior.

I am on Mac OS

**Comment by matthiasrebbe** [comment]
Mark,
by any chance, is it possible that for mouseup there  is something left here
_group "GLX2 Default Handler" of Cd "defaults main cd" of stack "revGLX2Defaults"_ 

and here

_button "my default handler" of cd "glx2Behaviors" of stack "revGLX2 default handlers master"_ 

At least in both scripts there is a preconfigured mouseup handler like this

```
on mouseUp pMouseBtnNum
   -- replace this with your own code
   throw param(0) && "has not yet been 
end mouseUp
```

while the other mouse.. handlers in those scripts are empty.

**Comment by mwieder** [comment]
Thanks, yes. Fixed now. That was leftover test code, and very specific to just new mouseUp handlers.

**Comment by matthiasrebbe** [comment]
The new version 4.12 build 2020.05.24 still adds the testcode  to new mouseup handlers.

**Comment by mwieder** [comment]
<groan> ok - I'll go look at that again. I probably overwrote something.

**Comment by mwieder** [comment]
Yep - I did indeed overwrite something. Will be fixed soon.

**Comment by mwieder** [comment]
As far as I can tell, this is fixed now with version 4.1.7.
